firmware/library/L2_HAL/temperature.hpp

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <cstdint>
+#include <cstdio>
+#include "L1_Drivers/i2c.hpp"
+
+class TemperatureInterface
+{
+ public:
+  virtual bool Initialize()                       = 0;
+  virtual int GetCelsius()                        = 0;
+  virtual int GetFahrenheit(int temperature_data) = 0;
+};
+
+class Temperature : public TemperatureInterface
+{
+ public:
+  bool Initialize() override
+  {
+  temperature_device_.Initialize();
+  uint8_t device_id = 0x00;
+  // Checks to see if the value read by I2c
+  // for the Temperature Chip ID register
+  // is the same as the Base ID (0x14).
+  temperature_device_.WriteThenRead(kTemperatureAddress, &kTemperatureId,
+  1, &device_id, 1);
+  return(device_id == kMemoryMapAddress);
+  }
+
+  int GetCelsius() override
+  {
+  uint8_t const kOneBurst [2]   = { 0xC4, 0x04 };
+  uint8_t const kAutoincBit [2] = { 0xC5, 0x01 };
+  // The register will enable the device to collect data once
+  // and automatically sets the stop bit to 0 (2nd bit).
+  status_ = temperature_device_.Write(kTemperatureAddress, kOneBurst, 2);
+  // Auto increments I2c register address pointer.
+  status_ = temperature_device_.Write(kTemperatureAddress, kAutoincBit, 2);
+  uint8_t least_significant_register = 0xC2;
+  uint8_t most_significant_register  = 0xC1;
```

Comment:
Make these into constexprs at the top of the method.

Reply:
Update on this portion. I left these as is because I2c did not like them as constexprs. That would make sense since the values are being over written.